pkg/api/legacyscheme/scheme.go

```diff
@@ -26,7 +26,7 @@ import (
 // extensions group instead. This Scheme is special and should appear ONLY in
 // the api group, unless you really know what you're doing.
 // TODO(lavalamp): make the above error impossible.
-var Scheme = runtime.NewScheme()
+var Scheme = runtime.NewNamedScheme("LegacyScheme")
```

Comment:
do we really want these strings? Or just use the filename from the strack everywhere? Wouldn't the later be enough? Would give less ugliness and less chances for error on copy&paste.